unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarNamesHUD/AvatarNamesTracker.cs

```diff
@@ -0,0 +1,82 @@
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace AvatarNamesHUD
+{
+    public class AvatarNamesTracker
+    {
+        private static readonly Vector3 OFFSET = new Vector3(0, 3f, 0);
+        private static readonly int VOICE_CHAT_ANIMATOR_TALKING = Animator.StringToHash("Talking");
+        private const float NAME_VANISHING_POINT_DISTANCE = 15.0f;
+
+        internal readonly RectTransform canvasRect;
+        internal readonly Image background;
+        internal readonly CanvasGroup backgroundCanvasGroup;
+        internal readonly TextMeshProUGUI name;
+        internal readonly CanvasGroup voiceChatCanvasGroup;
+        internal readonly Animator voiceChatAnimator;
+
+        internal PlayerStatus playerStatus;
+        internal bool visibility = false;
+
+        public AvatarNamesTracker(RectTransform canvasRect, RectTransform backgroundRect, RectTransform nameRect, RectTransform voiceChatRect)
+        {
+            this.canvasRect = canvasRect;
+            backgroundCanvasGroup = backgroundRect.GetComponent<CanvasGroup>();
+            background = backgroundRect.GetComponent<Image>();
+            name = nameRect.GetComponent<TextMeshProUGUI>();
+            voiceChatCanvasGroup = voiceChatRect.GetComponent<CanvasGroup>();
+            voiceChatAnimator = voiceChatRect.GetComponent<Animator>();
+        }
+
+        public void SetVisibility(bool visible)
+        {
+            visibility = visible;
+            background?.gameObject.SetActive(visibility);
```

Comment:
We shouldn't be using null-conditional operator on `UnityEngine.Object` instances as they can raise the `missingReferenceException` in weird cases. This is because the equality operator is overriden on that type, and the operator skips the override.

Reply:
Oh, I will use a manual check then, thx!

Reply:
In the end I decided to remove the null check and rely on the tests to prevent that edge case